### PR TITLE
Do not create lists if not necessary, use generator expressions

### PR DESCRIPTION
**Pull request by jakobj**
This PR removes three occurences where we created lists, while generator expressions are sufficient.

### CODE REVIEW

File: dicthash/dicthash.py [hunk 1]
```diff
@@ -68,10 +67,9 @@ def _generate_string_from_dict(d, blacklist, whitelist, prefix=''):
         whitelist = d.keys()
 
     if blacklist is not None:
-        whitelist = [key for key in whitelist if key not in blacklist]
+        whitelist = (key for key in whitelist if key not in blacklist)
 
-    raw = [_unpack_value(d[key], prefix + unicode(key)) for key in sorted(whitelist)]
-    return ''.join(raw)
+    return ''.join(_unpack_value(d[key], prefix + unicode(key)) for key in sorted(whitelist))
```
Comment (mschmidt87) [inline dicthash/dicthash.py]: It is unrelated to the topic of the PR, but we should still change this:
`prefix` is a keyword argument to `_unpack_value` so we should use it like this. Thus, please insert a `prefix=prefix+...` here.

**Review by mschmidt87 — CHANGES_REQUESTED** [review]
Very elegant. Just one minor request.

**Comment by jakobj** [comment]
thanks for the comment. I've now rebased onto master and addressed the issue you raised.

File: dicthash/dicthash.py [hunk 2]
```diff
@@ -71,10 +70,10 @@ def _generate_string_from_dict(d, blacklist, whitelist, prefix=''):
     if whitelist is None:
         whitelist = d.keys()
     if blacklist is not None:
-        whitelist = [key for key in whitelist if key not in blacklist]
+        whitelist = (key for key in whitelist if key not in blacklist)
+
     # Sort whitelist according to the keys converted to str
-    raw = [_unpack_value(d[key], prefix + str(key)) for key in sorted(whitelist, key=str)]
-    return ''.join(raw)
+    return ''.join(_unpack_value(d[key], prefix + str(key)) for key in sorted(whitelist, key=str))
```
Comment (mschmidt87) [inline dicthash/dicthash.py]: It is unrelated to the topic of the PR, but we should still change this:
prefix is a keyword argument to _unpack_value so we should use it like this. Thus, please insert a prefix=prefix+... here.

**Review by mschmidt87 — COMMENTED** [review]
I can't judge what you changed in the last round, but my complaint remains, see line comment.

**Comment by jakobj** [comment]
Haha, apparently nothing. Wait a second...

**Comment by jakobj** [comment]
I think I now managed to finally include your comment. ;)

**Review by mschmidt87 — APPROVED** [review]

**Pull request merged**